homeassistant/components/homeassistant_hardware/silabs_multiprotocol_addon.py

```diff
@@ -55,15 +59,64 @@
 SAVE_DELAY = 10
 
 
-@singleton(DATA_ADDON_MANAGER)
-async def get_addon_manager(hass: HomeAssistant) -> MultiprotocolAddonManager:
+@singleton(DATA_MULTIPROTOCOL_ADDON_MANAGER)
+async def get_multiprotocol_addon_manager(
+    hass: HomeAssistant,
+) -> MultiprotocolAddonManager:
     """Get the add-on manager."""
     manager = MultiprotocolAddonManager(hass)
     await manager.async_setup()
     return manager
 
 
-class MultiprotocolAddonManager(AddonManager):
+class SyncAddonManager(AddonManager):
+    """Addon manager which supports synchronous operations for managing an addon."""
+
+    async def async_wait_until_addon_state(self, *states: AddonState) -> None:
```

Comment:
This method could be generally useful in the parent addon manager. We can consider moving it in a follow up.